docs/source/deployment/amazon_emr_serverless.md

```diff
@@ -0,0 +1,244 @@
+across the entire setup but also enables control over the runtime environment.
+
+This approach helps to avoid job failures due to misconfigurations or potential conflicts with system-level packages,
+and ensures the package is portable so it can be debugged and tested locally ([see more details on validation](#optional-validate-the-custom-image)). Using the approach can provide a repeatable, reliable environment and improve operational flexibility.
+
+With this context established, the rest of this page describes how to deploy a Kedro project to EMR Serverless.
+
+## Overview of Approach
+
+This approach creates a custom Docker image for EMR Serverless to package dependencies and manage the runtime environment, and follows these steps:
+
+- Packaging the Kedro project to install it on all EMR Serverless worker nodes. `kedro package` can be used for this, and the resultant `.whl` file is used for installation.
+- Using a custom Python version instead of the default Python installed on EMR Serverless. In the example, [pyenv](https://github.com/pyenv/pyenv) is used for installing the custom Python version.
+- Running a job on EMR Serverless specifying Spark properties. This is needed to use the custom Python and provide an entrypoint script that accepts command line arguments for running Kedro.
+
+Here is an example Dockerfile that can be used:
+
+```shell
+FROM public.ecr.aws/emr-serverless/spark/emr-6.10.0:latest AS base
+
+USER root
+
+# Install Python build dependencies for pyenv
+RUN yum install -y gcc make patch zlib-devel bzip2 bzip2-devel readline-devel  \
+        sqlite sqlite-devel openssl11-devel tk-devel libffi-devel xz-devel tar
+
+# Install git for pyenv installation
+RUN yum install -y git
+
+# Add pyenv to PATH and set up environment variables
+ENV PYENV_ROOT /usr/.pyenv
+ENV PATH $PYENV_ROOT/shims:$PYENV_ROOT/bin:$PATH
+ENV PYTHON_VERSION=3.9.16
+
+# Install pyenv, initialize it, install desired Python version and set as global
+RUN curl https://pyenv.run | bash
+RUN eval "$(pyenv init -)"
+RUN pyenv install ${PYTHON_VERSION} && pyenv global ${PYTHON_VERSION}
+
+# Copy and install packaged Kedro project
+ENV KEDRO_PACKAGE <PACKAGE_WHEEL_NAME>
+COPY dist/$KEDRO_PACKAGE /tmp/dist/$KEDRO_PACKAGE
+RUN pip install --upgrade pip && pip install /tmp/dist/$KEDRO_PACKAGE  \
+    && rm -f /tmp/dist/$KEDRO_PACKAGE
+
+# Copy and extract conf folder
+ADD dist/conf.tar.gz /home/hadoop/
+
+# Initialize log files
+RUN mkdir -p /home/hadoop/logs
+RUN touch /home/hadoop/logs/info.log
+RUN touch /home/hadoop/logs/errors.log
+RUN chmod -R a+w /home/hadoop/logs
+
+# EMRS will run the image as hadoop
+USER hadoop:hadoop
+```
+
+Make sure to replace `<PACKAGE_WHEEL_NAME>` with your own `.whl` file name.
+Here is the `entrypoint.py` entrypoint script:
+
+```python
+import sys
+from <PACKAGE_NAME>.__main__ import main
+
+main(sys.argv[1:])
+
+```
+
```

Comment:
Can we use `python -m <python_package>` to replace this extra entrypoint file? It's just calling the main function and doesn't seem necessary.

Reply:
Resolving this as you've found the answer below. 😄